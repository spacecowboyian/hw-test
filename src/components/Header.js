import React from 'react';
import { Row, Col } from 'react-flexbox-grid';
import { css } from 'emotion/macro';
import { FiBell, FiArrowLeft } from "react-icons/fi";

import { colorCyan, spacing, standardBox } from '../Styles';
import Avatar from './Avatar';

const header = css`
 ${standardBox}
 text-align: left;
`;

const headerBackButton = css`
  display: flex;
  align-items: center;
  padding-left: 0px;
`;

const headerEmployeeInfo = css`
  display:flex;
  justify-content: flex-end;
  align-items: center;
`;

const headerEmployeeInfoName = css`
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  color: #006667;
  margin-right: calc(${spacing}*4);
`;

const iconLeftArrow = css`
  font-size: 2.2em;
  color: ${colorCyan};
`;

const iconNotificationBell = css`
  color: ${colorCyan};
  font-size: 2em;
`;

export default function Header({ user }) {
  const {
    initials,
    name
  } = user;
  return (
    <Row className={header}>
      <Col className={headerBackButton} xs={1}><FiArrowLeft className={iconLeftArrow} /></Col>
      <Col xs={11} className={headerEmployeeInfo}>
        <div className={headerEmployeeInfoName}>{name}</div>
        {Avatar(initials)}
        <FiBell className={iconNotificationBell} />
      </Col>
    </Row>
  )
}
