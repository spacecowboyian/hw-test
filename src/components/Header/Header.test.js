import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Header from './Header';

afterEach(cleanup);

const userData = {
  initials: 'ES',
  name: 'Elwin Sharvill',
}

it('renders correctly', () => {
  const tree = renderer
    .create(<Header user={userData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
