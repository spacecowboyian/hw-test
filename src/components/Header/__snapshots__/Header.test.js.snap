// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1ewjgy1-standardBox-header row"
>
  <div
    className="col-xs-1 css-1oe12ba-headerBackButton"
  >
    <svg
      className="css-1358zhw-iconLeftArrow"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="19"
        x2="5"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 19 5 12 12 5"
      />
    </svg>
  </div>
  <div
    className="col-xs-11 css-1tj8fi8-headerEmployeeInfo"
  >
    <div
      className="css-1gb5llw-headerEmployeeInfoName"
    >
      Elwin Sharvill
    </div>
    <div
      className="css-tkappr-avatarCircle"
    >
      <div
        className="css-bl0pin-avatarInitials"
      >
        ES
      </div>
    </div>
    <svg
      className="css-kt2y8z-iconNotificationBell"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 8A6 6 0 0 0 6 8c0 7-3 9-3 9h18s-3-2-3-9"
      />
      <path
        d="M13.73 21a2 2 0 0 1-3.46 0"
      />
    </svg>
  </div>
</div>
`;
